source/developers-guide/plugin-extension-by-plugin/index.md

```diff
@@ -0,0 +1,748 @@
+---
+layout: default
+title: Enhance a plugin with a plugin
+github_link: developers-guide/plugin-extendion-by-plugin/index.md
+indexed: true
+tags:
+  - extend
+  - plugin
+  - custom
+  - products
+---
+
+<div class="toc-list"></div>
+
+## Introduction
+This document will describe how existing plugins can be extended.
+
+With the objective of implement our own option type in CustomProducts which we can use in a Template and make it possible for a customer to upload files with special mime types.
+
+To achieve the objective we decorate a service to extend the function, extend ExtJs by Smarty blocks and implement new ExtJs and Smarty templates.
+
+
+In this wiki article we expand the plugin Custom Product (v2) with the following functions on base of the 5.2 plugin system: 
```

Comment:
On base of the new 5.2 plugin system, we extend the plugin SwagCustomProducts with the following functions.